src/mame/handheld/hh_ht11xx.cpp

```diff
@@ -0,0 +1,118 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+
+#include "emu.h"
+
+#include "cpu/ht1130/ht1130.h"
+
+#include "screen.h"
+#include "speaker.h"
+
+#define VERBOSE (0)
+#include "logmacro.h"
+
+namespace {
+
+class ht11xx_brickgame_state : public driver_device
+{
+public:
+	ht11xx_brickgame_state(const machine_config &mconfig, device_type type, const char *tag) :
+		driver_device(mconfig, type, tag),
+		m_maincpu(*this, "maincpu"),
+		m_out_x(*this, "seg%u_%u", 0U, 0U),
+		m_in1(*this, "IN1"),
+		m_in2(*this, "IN2")
+	{ }
+
+	void ht11xx_brickgame(machine_config &config);
+
+	DECLARE_INPUT_CHANGED_MEMBER(power_pressed);
+
+protected:
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+
+private:
+	void mcfg_svg_screen(machine_config &config, u16 width, u16 height, const char *tag = "screen");
+
+	void display_data_w(offs_t offset, u8 data);
+
+	required_device<ht1130_device> m_maincpu;
+	output_finder<256, 4> m_out_x;
+	required_ioport m_in1;
+	required_ioport m_in2;
+};
+
+void ht11xx_brickgame_state::machine_start()
+{
+	m_out_x.resolve();
+}
+
+void ht11xx_brickgame_state::machine_reset()
+{
+	for (int i=0;i<256;i++)
+		for (int j=0;j<4;j++)
+			m_out_x[i][j] = 0;
+}
+
+INPUT_CHANGED_MEMBER(ht11xx_brickgame_state::power_pressed)
+{
+	m_maincpu->wake();
+}
+
+static INPUT_PORTS_START( ht11xx_brickgame )
+	PORT_START("IN1")
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_START1 )
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_BUTTON5 ) PORT_NAME("Mute")
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_BUTTON6 ) PORT_NAME("Power") PORT_CHANGED_MEMBER(DEVICE_SELF, ht11xx_brickgame_state, power_pressed, 0) PORT_IMPULSE(1)
+
+	PORT_START("IN2") // not a joystick, but buttons are used for directional inputs in the snake game etc.
+	PORT_BIT( 0x01, IP_ACTIVE_LOW, IPT_BUTTON1 ) PORT_NAME("Up / Rotate")
+	PORT_BIT( 0x02, IP_ACTIVE_LOW, IPT_BUTTON2 ) PORT_NAME("Down / Drop")
+	PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_BUTTON3 ) PORT_NAME("Right")
+	PORT_BIT( 0x08, IP_ACTIVE_LOW, IPT_BUTTON4 ) PORT_NAME("Left")
+INPUT_PORTS_END
+
+
+void ht11xx_brickgame_state::display_data_w(offs_t offset, u8 data)
+{
+	for (int i = 0; i < 4; i++)
+	{
+		m_out_x[offset+0xb0][i] = (data >> i) & 1;
+	}
```

Comment:
What’s the fixed offset of 0xb0 here for?  Do those outputs actually exist?  Why instantiate them if they aren’t going to be used?

Reply:
the layout from the other emulator uses the memory address, the display RAM starts at 0xb0, this made the conversion process MUCH smoother, and less prone to introducing errors.  I'd rather not have to change it right now

Reply:
But you’re instantiating over 700 outputs that don’t exist.  That’s really ugly.  If you want the numbering to start at a different point, why don’t you use the actual output finder constructor argument that’s there for that?

Reply:
The prefered method is to match the COM and SEG pins, eg. x.y where X=0-3 for COM, y=0-31 for SEG (or however many segments this chip has).